src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -38,8 +43,21 @@ export const useTradeExecution = ({
     selectPortfolioAccountMetadataByAccountId(state, { accountId: sellAssetAccountId }),
   )
 
+  const buyAssetUsdRate = useAppSelector(state =>
+    selectUsdRateByAssetId(state, tradeQuote?.steps[0].buyAsset.assetId ?? ('' as AssetId)),
+  )
+
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetById(state, tradeQuote?.steps[0].sellAsset.assetId ?? ''),
+  )
+  const feeAssetUsdRate = useAppSelector(state =>
+    selectUsdRateByAssetId(state, feeAsset?.assetId ?? ('' as AssetId)),
```

Comment:
```suggestion
    feeAsset ? selectUsdRateByAssetId(state, feeAsset.assetId),
```

Reply:
Made some improvements to adopt this line of thinking, and use the proper selectors in https://github.com/shapeshift/web/pull/4847/commits/c74339b12b9b75e9f5930866150318945d88049c.